### ISSUE TITLE
Open different terminal from Docker Desktop on Windows 

### ISSUE THREAD

**Issue by nebuk89** [docker/roadmap#212]
Tell us about your request
When clicking the Cli button the default terminal should open the shell session for Windows users 

Which service(s) is this request for?
Docker Desktop on Windows 

Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?
Today customers have preferred terminals to work in which are setup to work for them, the desktop UI today opens the built in terminal for the machine not the users current default terminal.

Are you currently working around the issue?
Using the new MSFT terminal and not using the button in Desktop UI

Additional context:

Mac ticket: https://github.com/docker/roadmap/issues/98

**Comment by Kazmirchuk** [comment]
+1 this would be very handy, especially for novice users of Docker

**Comment by stephen-turner** [comment]
NB It looks like Microsoft will be introducing the notion of a "default terminal" into Windows: microsoft/terminal#492. I haven't seen an official statement, but I'm guessing it will land in 21H2.

**Comment by Lug-As** [comment]
+1

**Comment by dustypomerleau** [comment]
+1 on this. If you use the integrated terminal on an editor, like VS Code, but you also have iTerm2 installed, Docker Desktop will launch iTerm2, regardless of any settings. Thanks.

**Comment by YiuTerran** [comment]
+1

**Comment by dmelinosky** [comment]
This would be great +1

**Comment by MatthewOrganic** [comment]
+1

**Comment by Jerryporter** [comment]
+1

**Comment by hichemfantar** [comment]
> NB It looks like Microsoft will be introducing the notion of a "default terminal" into Windows: microsoft/terminal#492. I haven't seen an official statement, but I'm guessing it will land in 21H2.

It'd be preferable not to rely on windows default terminal because that's only customizable in w11. W10 users don't have that luxury.
It'd be better to be able to manually tell docker desktop our preferred terminal app.

**Comment by gerneio** [comment]
4.12.0 just released and initially I thought it had this feature based on the [release notes](https://docs.docker.com/desktop/release-notes/#new:~:text=Implemented%20an%20integrated%20terminal%20for%20containers.), but rather it was a feature for an "embedded" terminal within docker desktop itself, which is still pretty cool. So looks like we are close to getting this added, just need an option under the "Choose container terminal" docker desktop setting that does exactly that and lets you specify the system terminal to open. Either a free entry textbox that allows you to fill in the command to pass along to the auto-generated "docker exec" command or a dropdown of system defaults, similar to how the new windows terminal does it when opening its startup settings.

![image](https://user-images.githubusercontent.com/13949170/188278106-7445371d-e849-41db-b81e-0efd51839b58.png)

![image](https://user-images.githubusercontent.com/13949170/188278034-1f671420-e794-442f-909e-6aa770956d26.png)

**Comment by thoni56** [comment]
Although I wouldn't mind the automation, a much more easy stop-gap would be to allow copying of the docker command to use. I'm imagining a copy symbol just beside the "open in external terminal" which puts "docker exec ......" in to the clipboard.

Then I can just open my favourite terminal and paste.

This would work on all platforms (partly) solving also #395.

**Comment by joe0BAB** [comment]
Hi everyone here! I know this doesn't solve the feature request, but we are working on a new command that will make getting a shell into any container much easier. As of now, this isn't part of core Docker Desktop yet but distributed as an extension: [Docker Labs Debug Tools](https://open.docker.com/extensions/marketplace?extensionId=docker/labs-debug-tools-extension)

In short, with this you can do:
```
dld shell <your container or image>
```
I.e., it will give an easy to remember / short command that you can use from any terminal you want and will give you a shell into any image or container (including stopped), local or remote.

Related roadmap issue: #524 

Any feedback appreciated!

**Comment by gerneio** [comment]
@joe0BAB would this be something integrated into the docker desktop UI? I think that was the goal from the OP (and also what I'm looking for). In short, if we had the ability to specify the custom command that is opened/copied from docker desktop so we can then specify our preferred terminal, that would be great.

**Comment by joe0BAB** [comment]
@gerneio yes, eventually the idea is to integrate this new functionality that comes with the "Docker Labs Debug Tools" extension with the UI too

**Comment by thoni56** [comment]
And eventually that would become a shell command? Something like

    docker shell <container or image>

I like that.

**Comment by hichemfantar** [comment]
Looks like the latest Windows Terminal release allows you to set the windows terminal as your default terminal which means Docker Desktop should open up windows Terminal if it's your default.
![image](https://github.com/docker/roadmap/assets/34947993/d3eb96f0-26e3-4051-b9bb-39e422b81257)

**Comment by benitogf** [comment]
@hichemfantar  tested and works on:
![image](https://github.com/docker/roadmap/assets/1633813/72341b01-ecad-4171-b103-2f8c8d4bf088)
![image](https://github.com/docker/roadmap/assets/1633813/eefa3fdb-561b-4b3c-8302-786975364f64)

👍

**Comment by Daviid-P** [comment]
Is this actually solved as [benitogf](https://github.com/benitogf) claims?

I have the default set to Windows Terminal inside Terminal > Configuration > Default Terminal Application but regardless of what I set in Docker Desktop Settings, Integrated or System Default I get the same plain CMD windows with `docker exec -it ........` as the title.

Open in external terminal and the button next to the stop button do the same thing.


```
You're up to date

Docker Desktop 4.28.0 (139021) is currently the newest version available.
```

**Comment by TheFeloniousMonk** [comment]
Checking in on this as well. Every time I open the terminal I get the default windows CMD, and have to close it and open BASH. Opening new terminals, I have to use the dropdown and select BASH every time. It's very inefficient, and clearly this has been an open user request for years now.
